File: backend/debug_login.py
#!/usr/bin/env python
"""
Debug login response
"""

import requests

BASE_URL = "http://127.0.0.1:8000/api"

def debug_login():
    login_url = f"{BASE_URL}/auth/login/"
    login_data = {
        "email": "testadmin@example.com",
        "password": "testpass123"
    }
    
    try:
        response = requests.post(login_url, json=login_data)
        print(f"Status Code: {response.status_code}")
        print(f"Headers: {dict(response.headers)}")
        print(f"Raw Response Text: '{response.text}'")
        print(f"Response Length: {len(response.text)}")
        
        if response.text.strip():
            try:
                json_data = response.json()
                print(f"JSON Data: {json_data}")
            except Exception as e:
                print(f"JSON parsing error: {e}")
        else:
            print("Empty response body")
            
    except Exception as e:
        print(f"Request error: {e}")

if __name__ == "__main__":
    debug_login()

File: frontend/public/debug-login.js
// Login shortcut for testing
document.addEventListener('DOMContentLoaded', function() {
  const loginForm = document.createElement('div');
  loginForm.className = 'login-debug-form';
  loginForm.style.position = 'fixed';
  loginForm.style.bottom = '10px';
  loginForm.style.right = '10px';
  loginForm.style.background = '#f0f0f0';
  loginForm.style.padding = '10px';
  loginForm.style.border = '1px solid #ccc';
  loginForm.style.zIndex = '1000';

  loginForm.innerHTML = `
    <h3>Debug Login</h3>
    <input id="debug-email" type="text" placeholder="Email" value="testuser@example.com" />
    <input id="debug-password" type="password" placeholder="Password" value="password123" />
    <button id="debug-login">Login</button>
  `;

  document.body.appendChild(loginForm);

  document.getElementById('debug-login').addEventListener('click', function() {
    const email = document.getElementById('debug-email').value;
    const password = document.getElementById('debug-password').value;
    
    // Make login request
    fetch('http://localhost:8000/api/auth/login/', {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        'Accept': 'application/json'
      },
      body: JSON.stringify({ email, password })
    })
    .then(response => response.json())
    .then(data => {
      console.log('Login response:', data);
      if (data.access) {
        // Save tokens and user data
        localStorage.setItem('accessToken', data.access);
        localStorage.setItem('refreshToken', data.refresh);
        
        // Parse the JWT to get user info
        const payload = JSON.parse(atob(data.access.split('.')[1]));
        localStorage.setItem('user', JSON.stringify({
          id: payload.user_id,
          email: payload.email,
          username: payload.username,
          role: payload.role || 'customer'
        }));
        
        alert('Login successful!');
        location.reload();
      } else {
        alert('Login failed: ' + (data.detail || 'Unknown error'));
      }
    })
    .catch(error => {
      console.error('Login error:', error);
      alert('Login error: ' + error.message);
    });
  });
});